tests/ApiProblemTest.php

```diff
@@ -42,6 +52,27 @@ public function testComplexExtraProperty() : void
         $this->assertEquals('Zim', $problem['irken']['invader']);
     }
 
+    public function testSimpleJsonCompileWithJsonException() : void
+    {
+        $this->expectException(JsonEncodeException::class);
+        $this->expectExceptionCode(\JSON_ERROR_UTF8);
+        $this->expectExceptionMessage('Malformed UTF-8 characters, possibly incorrectly encoded');
+
+        // an invalid string value
+        $problem = new ApiProblem(\hex2bin('58efa99d4e19ff4e93efd93f7afb10a5'), 'URI');
+
+        $json = $problem->asJson();
+        $result = json_decode($json, true);
```

Comment:
Isn't everything past this line redundant, since if an exception isn't thrown here the test has failed anyway?

Reply:
You're right, I forgot to remove last statements